cfg/types.go

```diff
@@ -85,10 +86,17 @@ func NewDefaultConfig() *Configuration {
 	}
 }
 
-//CoreObjectInterface defines interface for core kubernetes resources
-type CoreObjectInterface interface {
-	Delete(name string, options *metav1.DeleteOptions) error
+func (c *Configuration) KubernetesNamespace() string {
+	if c.Namespace == "" {
+		log.Debug("Determining namespace from kubeconfig...")
+		defaultNamespace := "default"
+		namespace, err := kubernetes.Namespace()
+		if err != nil {
+			log.Warningf("Unable to determine default namespace. Falling back to '%s'", defaultNamespace)
+			c.Namespace = defaultNamespace
```

Comment:
I'm not sure if this is a good idea. Isn't the "default" namespace a privileged namespace? Couldn't falling back to it as a default potentially have catastrophic effects?

What if we require the user to log in to the cluster and simply abort here?

Reply:
On a plain (e.g. Rancher) K8s cluster, the default is empty and not at all "privileged". It's only OpenShift 3.x where Red Hat decided to run the registry and router in it (which makes no sense but not the discussion item here). Users don't have access, there's nothing dangerous in using this. It's only dangerous for cluster administrators.

I didn't introduce this default. Personally I agree with you, I would make the namespace in the config (and implied  authorization) mandatory and not default to another namespace.

What was the reason for defaulting to "default" namespace @ZuGao ?

Reply:
No one benefits from falling back to "default" namespace, because in 100% of the cases, we want to cleanup stuff on a certain application namespace, not in a fallback

Reply:
I think the fallback was to make the tests pass in the pipeline. But that should be obsolete, we do mock the namespace now, don't we?

Reply:
I don't know, hard to imagine since the `Namespace` property in the `Configuration` property is just a string, so should be easily modifiable for testing purposes.

Reply:
> What was the reason for defaulting to "default" namespace @ZuGao ?

As @bittner it was for testing purposes so I also don't think it's a good idea to use it.

> I think the fallback was to make the tests pass in the pipeline. But that should be obsolete, we do mock the namespace now, don't we?

The introduction of gomock in the project is still ongoing but yes it will become obsolete

Reply:
Let's talk about that in #35

Reply:
Current behaviour:
* Order of namespace precedence: `ARG` > `--namespace` flag > namespace from kubeconfig > error
* There is no default namespace. If it cannot be determined, it errors out.
* we don't need to "mock" this in the tests anymore

I think we can close this discussion